src/fparser/block_statements.py

```diff
@@ -827,22 +827,50 @@ class EndSelect(EndStatement):
 
 class Select(BeginStatement):
     """
-    [ <case-construct-name> : ] SELECT CASE ( <case-expr> )
+    Base class for the Select (case/type) statement
 
     """
     match = re.compile(r'select\s*case\s*\(.*\)\Z',re.I).match
     end_stmt_cls = EndSelect
     name = ''
-    def tostr(self):
-        return 'SELECT CASE ( %s )' % (self.expr)
+
     def process_item(self):
-        self.expr = self.item.get_line()[6:].lstrip()[4:].lstrip()[1:-1].strip()
+        ''' Populate the state of this Select object by parsing the
+        associated line of code '''
+        self.expr = self.item.get_line()[6:].lstrip()[4:].\
+                    lstrip()[1:-1].strip()
         self.construct_name = self.item.name
         return BeginStatement.process_item(self)
 
     def get_classes(self):
         return [Case] + execution_part_construct
 
+
+class SelectCase(Select):
+    """
+    [ <case-construct-name> : ] SELECT CASE ( <case-expr> )
+
+    """
+    match = re.compile(r'select\s*case\s*\(.*\)\Z', re.I).match
+
+    def tostr(self):
+        return 'SELECT CASE ( %s )' % (self.expr)
+
```

Comment:
Again, well spotted. I've moved `get_classes()` out of the base class and into the `SelectCase` class where it belongs.